Sources/TuistGenerator/Generator/ProjectGenerator.swift

```diff
@@ -248,12 +259,23 @@ final class ProjectGenerator: ProjectGenerating {
                                                    generatedProject: generatedProject)
     }
 
+    enum XcodeVersionError: Swift.Error {
+        case noXcode
+        case noVersion
+    }
+
     private func determineProjectConstants() -> ProjectConstants {
-        // TODO:
-        //
-        // To maintain backwards compatibility with Xcode 10
-        // the Xcode10 constants are used unless the use of Xcode 11
-        // features are detected (e.g Swift PM dependencies)
-        return .xcode10
+        do {
+            let version = try XcodeController(system: system).selectedVersion()
+
+            if version.major >= 11 {
+                return .xcode11
+            } else {
+                return .xcode10
+            }
+
+        } catch {
```

Comment:
If we can't obtain the Xcode version I'd error. Not erroring might result in unexpected results for the users that will be hard to debug.

Reply:
I think that's a proper reasoning, the error bubbles up now.